### ISSUE TITLE
Error 400 - Restrict-MDEIPAddress

### ISSUE THREAD

**Issue by tora1104** [Azure/Azure-Sentinel#4052]
Used the deploy to Azure button for alert trigger from [https://github.com/Azure/Azure-Sentinel/tree/master/Playbooks/Restrict-MDEIPAddress](https://github.com/Azure/Azure-Sentinel/tree/master/Playbooks/Restrict-MDEIPAddress).

Followed the instructions to add permissions - 
$MIGuid = "<Enter your managed identity guid here>"
$MI = Get-AzureADServicePrincipal -ObjectId $MIGuid

$MDEAppId = "fc780465-2017-40d4-a0c5-307022471b92"
$PermissionName = "Ti.ReadWrite" 

$MDEServicePrincipal = Get-AzureADServicePrincipal -Filter "appId eq '$MDEAppId'"
$AppRole = $MDEServicePrincipal.AppRoles | Where-Object {$_.Value -eq $PermissionName -and $_.AllowedMemberTypes -contains "Application"}
New-AzureAdServiceAppRoleAssignment -ObjectId $MI.ObjectId -PrincipalId $MI.ObjectId `
-ResourceId $MDEServicePrincipal.ObjectId -Id $AppRole.Id

Getting the following error:
  "error": {
    "code": 400,
    "source": "logic-apis-northcentralus.azure-apim.net",
    "clientRequestId": "10029ec8-a4dc-4760-a4c5-b7d208dd07b7",
    "message": "The response is not in a JSON format.",
    "innerError": "Invalid subscription id or resource group or API connection"

Check the subscription ID and resource group - both are correct. Not sure how to check the API connection but verified I ran the above to give the managed identity permissions.

What am I missing? 

Thanks for your help!

**Comment by dicolanl** [comment]
Hi @tora1104  which step threw this error?  could you share a screenshot?

**Comment by tora1104** [comment]
The Alert-Get Incident step erred out -

![image](https://user-images.githubusercontent.com/51537003/151823763-c4f92da8-e158-43d7-aff2-1f595fb08e8a.png)

![image](https://user-images.githubusercontent.com/51537003/151824217-720dbe72-47df-4144-a5e0-efd77832ed97.png)

**Comment by dicolanl** [comment]
This doesnt look like a permissions issue or issue with the API connection config per se.  I recommend you open a support ticket as it appears the connector is having an issue.

**Comment by samikroy** [comment]
@tora1104 & @dicolanl  - Was able to fix this by assigning Microsoft Sentinel Reader at Resource group for the MSI.
And have created PR #4077 to embed the script for the same.
Request you review .

![image](https://user-images.githubusercontent.com/20562985/152164459-f94b6f7c-baa0-475f-af63-e64c337e2e48.png)


![image](https://user-images.githubusercontent.com/20562985/152164563-d58153b6-fc06-49ab-8c1c-4a54137f9446.png)

**Comment by tora1104** [comment]
@samikroy Thanks for the info! 

I opened a ticket and was given similar instructions on giving the managed identity Microsoft Sentinel Reader permissions - but this was at the Sentinel level instead of the resource group. Not sure if that matters? I just tested and it worked. 

Here is the link they sent me - [https://techcommunity.microsoft.com/t5/microsoft-sentinel-blog/what-s-new-managed-identity-for-azure-sentinel-logic-apps/ba-p/2068204](https://techcommunity.microsoft.com/t5/microsoft-sentinel-blog/what-s-new-managed-identity-for-azure-sentinel-logic-apps/ba-p/2068204)

Was missing this part - 
![image](https://user-images.githubusercontent.com/51537003/152172053-cd6358f8-8764-4756-8e72-5dd27fb579bd.png)
